pkgs/development/libraries/qt-5/qtModule.nix

```diff
@@ -25,8 +25,14 @@ mkDerivation (args // {
 
   nativeBuildInputs =
     (args.nativeBuildInputs or []) ++ [
-      perl qmake
+      perl
+      # `qmake`(`pkgsBuildHost.qt5.qmake`) propagates `qtbase` which causes there to be
+      # `pkgsBuildHost.qt5.qtbase` and the `pkgsHostTarget.qt5.qtbase.dev` in the build environment
+      (if (stdenv.buildPlatform != stdenv.hostPlatform) then pkgsHostTarget.qt5.qmake else qmake)
     ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
+      # .dev is required for binaries
+      # qmake propagates `qtbase` but on cross it's in `depsTargetTargetPropagated` so it's treated as a `buildInput`
```

Comment:
Why does qmake propagate qtbase in this way?  That doesn't sound right…

Reply:
Because
```
$ nix build ".#pkgsCross.aarch64-multiplatform.qt5.qtbase^*"
$ file ./result-dev/bin/moc
./result-dev/bin/moc: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52/lib/ld-linux-x86-64.so.2, for GNU/Linux 3.10.0, not stripped
```

https://github.com/NixOS/nixpkgs/pull/267311 looks like a cleaner option

Reply:
> Why does qmake propagate qtbase in this way? That doesn't sound right…

Here are my rambling explanations
https://github.com/NixOS/nixpkgs/issues/269756#issuecomment-1837976928